python-avd/pyavd/_eos_designs/structured_config/base/snmp_server.py

```diff
@@ -60,26 +60,73 @@ def snmp_server(self: AvdStructuredConfigBaseProtocol) -> None:
             traps=snmp_settings.traps,
         )
 
+    def _snmp_engine_id_hash(self: AvdStructuredConfigBaseProtocol, management_ip: str) -> str:
+        # Accepting SonarLint issue: The weak sha1 is not used for encryption. Just to create a unique engine id.
+        digest = sha1(f"{self.shared_utils.hostname}{management_ip}".encode()).hexdigest()  # NOSONAR # noqa: S324
+        # prefix with Enterprise Id + 04 to adhere to RCF3411 and RFC5343
+        # Arista Enterprise ID = 30065 (7571 in hex)
+        # 5th octet = 04 , meaning engine id is based on custom text
+        return f"8000757104{digest}"
+
+    def _snmp_engine_ids_inband(self: AvdStructuredConfigBaseProtocol) -> str:
+        """Returns the SNMP engine id based on Inband Management."""
+        # TODO: add support for inband_mgmt_subnet calculation, would need calculation done in SharedUtils
+        # TODO: add support for custom inband_mgmt_interface, would need calculation done in SharedUtils
+        if not self.shared_utils.inband_mgmt_ip:
+            msg = "hostname_inband_ip engine id calculation requires inband_mgmt_ip"
```

Comment:
the new error message is `msg = f"'{context}' is set to 'use_inband_mgmt_vrf' but this node is missing configuration for inband management."` which may not be very relevant here.
 
```suggestion
            msg = "'snmp_settings.compute_local_engineid_source: hostname_inband_ip' engine id calculation requires 'inband_mgmt_ip'"
```